helix-core/src/main/java/org/apache/helix/controller/stages/ClusterEvent.java

```diff
@@ -55,6 +55,19 @@ public ClusterEvent(String clusterName, ClusterEventType eventType, String event
     _eventId = eventId;
   }
 
+  /**
```

Comment:
Thanks for the PR!

1. Can we just replace clone() logic with what you have in the copy constructor? If the constructor is private, then I think it'd just be simpler to put this "copy" logic in clone() (it seems pretty simple anyways).

2. With that said, do we still need addAttribute() anymore?

Reply:
That's really what I thought initially. But it turns out the variable `_eventAttributeMap` is `final`, which means I cannot mutate the value. 
Are you also referring to this
```
ClusterEvent newEvent = new ClusterEvent(_clusterName, _eventType, eventId);	    return new ClusterEvent(this, eventId);
newEvent.setCreationTime(_creationTime);
newEvent._eventAttributeMap = new HashMap(__eventAttributeMap);
```